File: go/Six.go
package main

import "fmt"
import "os"
import "bufio"
import "strings"
import "strconv"
import "time"

func main() {
	fmt.Println("What is your current age? ")
	reader := bufio.NewReader(os.Stdin)
	fst, _ := reader.ReadString('\n')
	fst = strings.TrimSpace(fst)
	a, _ := strconv.Atoi(fst)
	fmt.Println("At what age would you like to retire?  ")
	snd, _ := reader.ReadString('\n')
	snd = strings.TrimSpace(snd)
	b, _ := strconv.Atoi(snd)
	now := time.Now()
	fmt.Printf("You have %d years left until you can retire\n", b-a)
	fmt.Printf("It's %d so you can retire in %d.\n", now.Year(), now.Year()+b-a)
}

File: rust/six.rs
//Six.rs
//
//

use std::str::FromStr;
use std::time::{SystemTime,UNIX_EPOCH};

fn main() {
    println!("What is your current age?");
    let mut age = String::new();
    std::io::stdin().read_line(&mut age)
        .expect("Failed to read line");
    let the_age = i32::from_str(age.trim_right()).unwrap();

    println!("At what age would you like to retire?");
    let mut retire = String::new();
    std::io::stdin().read_line(&mut retire)
        .expect("Failed to read line");
    let the_retire = i32::from_str(retire.trim_right()).unwrap();

    let now = SystemTime::now();
    let mils = now.duration_since(UNIX_EPOCH).expect("time fail");
    println!("You have {} years left until you can retire.", the_retire - the_age);

    // I think the cannonical way is the crono crate?
    println!("It's {:?} so you can retire in {}.", mils, the_age );
    println!("It's {:?}.", now );

}    
